modules/oe_theme_helper/tests/src/Kernel/Plugin/Field/FieldFormatter/FeaturedMediaFormatterTest.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_helper\Kernel\Plugin\Field\FieldFormatter;
+
+use Drupal\Component\Utility\UrlHelper;
+use Drupal\field\Entity\FieldConfig;
+use Drupal\field\Entity\FieldStorageConfig;
+use Drupal\file\Entity\File;
+use Drupal\Tests\oe_theme\Kernel\AbstractKernelTestBase;
+use Drupal\node\Entity\Node;
+use Drupal\node\Entity\NodeType;
+
+/**
+ * Tests the featured media formatter.
+ */
+class FeaturedMediaFormatterTest extends AbstractKernelTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static $modules = [
+    'field',
+    'node',
+    'media',
+    'file',
+    'image',
+    'oe_media',
+    'oe_content_featured_media_field',
+    'system',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    $this->installSchema('file', 'file_usage');
+    $this->installEntitySchema('media');
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('file');
+    $this->installConfig([
+      'media',
+      'image',
+      'file',
+      'system',
+      'oe_media',
+      'oe_content_featured_media_field',
+    ]);
+
+    // Create a content type.
+    $type = NodeType::create(['name' => 'Test content type', 'type' => 'test_ct']);
+    $type->save();
+
+    FieldStorageConfig::create([
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'type' => 'oe_featured_media',
+      'cardinality' => 1,
+      'entity_types' => ['node'],
+    ])->save();
+
+    FieldConfig::create([
+      'label' => 'Featured media field',
+      'field_name' => 'featured_media_field',
+      'entity_type' => 'node',
+      'bundle' => 'test_ct',
+      'settings' => [
+        'handler' => 'default:media',
+        'handler_settings' => [
+          'target_bundles' => [
+            'image' => 'image',
+          ],
+        ],
+        'sort' => [
+          'field' => '_none',
+        ],
+        'auto_create' => '0',
+      ],
+      'required' => FALSE,
+    ])->save();
+  }
+
+  /**
+   * Test the featured media formatter.
+   */
+  public function testFormatter(): void {
+    \Drupal::service('file_system')->copy($this->root . '/core/misc/druplicon.png', 'public://example.jpg');
```

Comment:
We can use the available $this->container instead of retrieving all the services statically.

Reply:
because the containers can get out of sync, @brummbar can explain

Reply:
I thought the agreement was that we would not use the container on functional tests but this is a kernel test. Am I not remembering correctly?

Reply:
I cant remember, I'll ask @brummbar real quick :)